jsbits/diff.js

```diff
@@ -165,30 +168,34 @@ function syncChildren (os, ns, parent) {
     , nLast
     , oFirst
     , oLast
-    , LIS
-    , temp;
+    , tmp
+    , found
+    , node;
     for (;;) {
 	 /* check base case, first > last for both new and old
 	   [ ] -- old children empty (fully-swapped)
 	   [ ] -- new children empty (fully-swapped)
 	 */
-	if (newFirstIndex > newLastIndex && oldFirstIndex > oldLastIndex) break;
-	  nFirst = ns[newFirstIndex];
-	  nLast  = ns[newLastIndex];
-	  oFirst = os[oldFirstIndex];
-	  oLast  = os[oldLastIndex];
+	if (newFirstIndex > newLastIndex && oldFirstIndex > oldLastIndex) {
+	    break;
+	}
+
+	/* Initialize */
+	nFirst = ns[newFirstIndex];
+	nLast  = ns[newLastIndex];
+	oFirst = os[oldFirstIndex];
+	oLast  = os[oldLastIndex];
+
 	/* No more old nodes, create and insert all remaining nodes
 	   -> [ ] <- old children
 	   -> [ a b c ] <- new children
 	*/
 	if (oldFirstIndex > oldLastIndex) {
-	    var node = createNodeDontAppend(nFirst);
+	    createElement(nFirst);
 	    /* insertBefore's semantics will append a node if the second argument provided is `null` or `undefined`.
 	       Otherwise, it will insert node.domRef before oLast.domRef. */
-	    parent.insertBefore(node.domRef, oLast ? oLast.domRef : null);
-	    os.splice(newFirstIndex, 0, nFirst);
+            parent.insertBefore (nFirst.domRef, os[oldLastIndex] ? os[oldLastIndex].domRef.nextSibling : null);
```

Comment:
For consistency, this should either use `oLast`  or `oLast` should be removed.

Reply:
fancy :)